src/hdmf/build/classgenerator.py

```diff
@@ -0,0 +1,336 @@
+from copy import deepcopy
+from datetime import datetime
+
+import numpy as np
+
+from ..container import Container, Data, DataRegion, MultiContainerInterface
+from ..spec import AttributeSpec, LinkSpec, RefSpec
+from ..spec.spec import BaseStorageSpec, ZERO_OR_MANY, ONE_OR_MANY
+from ..utils import docval, getargs, ExtenderMeta, get_docval, fmt_docval_args
+
+
+class ClassGenerator:
```

Comment:
could this functionality just be part of TypeMap?

Reply:
Yes. The only benefit of having it outside TypeMap is to organize it as separate, isolated functionality, but there is so little code, that I'd be fine moving it to TypeMap.